shiipou/simplon-2024-brief-06-TooManyChoco
# build(gradle): gradle migration now find resources in .jar file to run migrations

## About this PR



Gradle didn't find the migrations scripts on the current project if we build the `jar` using `./gradle jar`
The produced `jar` is located in `java/build/libs/java.jar` and you can run it with `java -jar java/build/libs/java.jar`.

My change move the migrations script into the resources directory and change the DbMigrator implementation so it can read it and made everything else working the same way if we use `./gradlew run` or `java -jar java/build/libs/java.jar` command.


## Technical highlight/advice


## How to test my changes


1. Reset your database by dropping every tables showing in `\dt` when you're connected using `psql`.
2. Build the project using `./gradlew jar`
3. Run the produced jar with `java -jar  java/build/libs/java.jar`.
4. See migrations working.
5. Reset again your database like in step 1.
6. Run the project with ./gradlew run`.
7. See migrations working.

That's all.


## Checklist

### I included unit tests that cover my changes

### I added/updated the documentation about my changes
